src/codegen/mod.rs

```diff
@@ -3948,9 +3948,13 @@ impl CodeGenerator for Function {
         {
             attributes.push(attributes::must_use());
         }
-
+        let mut comment_ = None;
         if let Some(comment) = item.comment(ctx) {
-            attributes.push(attributes::doc(comment));
+            if ctx.options().dynamic_library_name.is_none() {
+                attributes.push(attributes::doc(comment));
```

Comment:
So this looks good as a one-off, but it seems to me we should also preserve stuff like the `#[must_use]` attribute, etc.

So perhaps we can do something like this instead:

```
let is_dynamic_function = ctx.options().dynamic_library_name.is_some() && self.kind() == FunctionKind::Function;
```

Then don't push the link_name etc attributes when we're not a dynamic function, and pass the whole attributes array to the dyngen. wdyt?

Reply:
I think that's a good alternative. I'll try to get a test working :+1:

Reply:
done!